src/store/tools/useAnnotationTool.ts

```diff
@@ -0,0 +1,162 @@
+import { Ref, computed, del, ref, set } from '@vue/composition-api';
+import { removeFromArray } from '@/src/utils';
+
+import { TOOL_COLORS } from '@/src/config';
+import { defineStore } from 'pinia';
+import { useCurrentImage } from '@/src/composables/useCurrentImage';
+import {
+  FrameOfReference,
+  frameOfReferenceToImageSliceAndAxis,
+} from '@/src/utils/frameOfReference';
+import { useViewStore } from '@/src/store/views';
+import { getLPSAxisFromDir } from '@/src/utils/lps';
+import { useViewConfigStore } from '@/src/store/view-configs';
+import { LPSAxisDir } from '@/src/types/lps';
+import { findImageID, getDataID } from '@/src/store/datasets';
+
+const useDependencyInjectionStore = defineStore('dependencyInjection', {});
+const useIDGenerator = () => {
+  const { $id } = useDependencyInjectionStore();
+  return $id;
+};
+
+type AnnotationTool = {
+  id: unknown;
```

Comment:
Is it possible to type this to `string`? `RectangleID extends string` should be true with that change, and same with `Rectangle extends AnnotationTool`. This means we can avoid the `ToolID extends Tool['id'] & string` below.

Reply:
Good idea.  Fixed.